cmd/fry_test.go

```diff
@@ -16,39 +16,84 @@
 package cmd
 
 import (
-	"strings"
 	"testing"
 
+	"github.com/jarcoal/httpmock"
 	"github.com/sonatype-nexus-community/hashbrowns/types"
 	"github.com/stretchr/testify/assert"
 )
 
+const applicationsResponse = `{
+	"applications": [
+		{
+			"id": "4bb67dcfc86344e3a483832f8c496419",
+			"publicId": "testapp",
+			"name": "TestApp",
+			"organizationId": "bb41817bd3e2403a8a52fe8bcd8fe25a",
+			"contactUserName": "NewAppContact",
+			"applicationTags": [
+				{
+					"id": "9beee80c6fc148dfa51e8b0359ee4d4e",
+					"tagId": "cfea8fa79df64283bd64e5b6b624ba48",
+					"applicationId": "4bb67dcfc86344e3a483832f8c496419"
+				}
+			]
+		}
+	]
+}`
+
+const thirdPartyAPIResultJSON = `{
+		"statusUrl": "api/v2/scan/applications/4bb67dcfc86344e3a483832f8c496419/status/9cee2b6366fc4d328edc318eae46b2cb"
+}`
+
+const pollingResult = `{
+	"policyAction": "None",
+	"reportHtmlUrl": "http://sillyplace.com:8090/ui/links/application/test-app/report/95c4c14e",
+	"isError": false
+}`
+
 func validateConfigFryError(t *testing.T, expectedErrorMsgSnippet string, expectedConfig types.Config, args ...string) {
 	_, err := executeCommand(rootCmd, args...)
 
-	if expectedErrorMsgSnippet != "" {
-		assert.Error(t, err)
-		assert.True(t, strings.Contains(err.Error(), expectedErrorMsgSnippet))
-	} else {
-		assert.NoError(t, err)
-	}
-
-	assert.Equal(t, expectedConfig, config)
+	assert.NotNil(t, err)
+	assert.Equal(t, expectedErrorMsgSnippet, err.Error())
```

Comment:
Do you really want to enforce the entire error string here? Previously only required the message "contain" the expected value:

```
assert.True(t, strings.Contains(err.Error(), expectedErrorMsgSnippet))
```

I'm seeing some environment specific diffs that might not be relevant, e.g.:

```
TestFryCommandConfigNoServerRunning: fry_test.go:59: 
        	Error Trace:	fry_test.go:59
        	            				fry_test.go:77
        	Error:      	Not equal: 
        	            	expected: "Get \"http://localhost:8070/api/v2/applications?publicId=testapp\": dial tcp 127.0.0.1:8070: connect: connection refused"
        	            	actual  : "Get \"http://localhost:8070/api/v2/applications?publicId=testapp\": dial tcp [::1]:8070: connect: connection refused"
```

tcp 127.0.0.1:8070
vs
tcp [::1]:8070

Reply:
Yeah I saw that failure (which is why I commented it out, originally). It might be better to just assert on type of error?

Reply:
I switched this test to mock the server, so now we control the actual response, etc...

In: edc20fb